### PR TITLE
[codex] Reach 100 tok/s Qwen3.6 Lucebox

### PR DESCRIPTION
**Pull request by DeanoC**
## Summary

This PR pushes the Qwen3.6-27B Lucebox-style DFlash path over the 100 tok/s mean target on the RX 7900 XTX / gfx1100 setup.

Key changes:

- Adds fused/raw-GGML small-M verifier optimizations carried through the 100 tok/s phase.
- Adds fused Q6_K m16 lm-head argmax for target and draft greedy paths to avoid full logits materialization where only token IDs are needed.
- Extends residual-add matmul epilogues to append attention output projections.
- Adds a default-on append recurrent warp32 direct-attention path that writes BF16 attention output directly, updates F32 recurrent state directly, and preserves the exact Q8 rollback trace.
- Adds GPU parity coverage for the append recurrent direct-attention path.
- Updates the Qwen3.6 Lucebox parity log with kept/rejected phases and benchmark artifacts so future profiling does not repeat stale probes.

Rollback gates added/kept for risky paths include:

- `SUPERSONIC_DFLASH_DISABLE_Q6_K_LM_HEAD_ARGMAX_FUSED=1`
- `SUPERSONIC_DFLASH_DISABLE_APPEND_RECURRENT_WARP32=1`
- existing residual/MLP rollback gates documented in the parity log

## Performance

Previous best before the final append recurrent direct path:

- `target/qwen36_100tok_profile2/append_attn_residual_fused_10x256.json`
- mean `97.12 tok/s`, weighted `95.75 tok/s`, generated `1654`, stopped early `10/10`

Confirmed best after this PR:

- `target/qwen36_100tok_profile2/append_recurrent_warp32_direct_10x256.json`
- mean `100.86 tok/s`, weighted `99.40 tok/s`, min `87.87`, max `118.20`, generated `1654`, stopped early `10/10`

Repeat run:

- `target/qwen36_100tok_profile2/append_recurrent_warp32_direct_10x256_rerun.json`
- mean `100.79 tok/s`, weighted `99.38 tok/s`, min `87.87`, max `117.92`, generated `1654`, stopped early `10/10`

Combined output hash stayed `032209e65467e8aa6c74025dc8b70b325f0ec767054ddff614e04550bb11f3bf`, and per-prompt generated-token counts stayed `[179,232,99,159,174,154,216,114,165,162]`.

The final FFI profile confirms the intended kernel swap:

- old `qwen.delta_recurrent_prefill_capture_q8_trace`: `199.52 ms`
- old `qwen.dflash_extract_recurrent_attn`: `31.39 ms`
- new `qwen.delta_recurrent_prefill_capture_q8_trace_attn`: `132.42 ms`

## Validation

- `cargo fmt --check`
- `git diff --check`
- `HIP_ARCH=gfx1100 cargo build --release --bin supersonic`
- `cargo test -p runner --test dflash_tree_delta_parity dflash_append_delta_q8_direct_attention_matches_extract_path --release -- --ignored --nocapture`
- one-prompt enabled/rollback-gated A/B for append recurrent warp32 direct path
- one-prompt FFI shape profile for append recurrent warp32 direct path
- full 10-prompt Lucebox-style suite twice with matching output hashes and generated-token counts

## Notes

The explicit mean-throughput goal is met twice. Weighted throughput is still just under 100 tok/s, so a future follow-up could target weighted >=100 if we want the stricter aggregate too.

### CODE REVIEW

**Comment by DeanoC** [comment]
@codex review
